concrete/src/Application/Application.php

```diff
@@ -198,7 +199,9 @@ public function handleAutomaticUpdates()
         $installed = $config->get('concrete.version_db_installed');
         $core = $config->get('concrete.version_db');
         if ($installed < $core) {
-            Update::updateToCurrentVersion();
+            $this->make(MutexInterface::class)->execute(Update::MUTEX_KEY, function() {
```

Comment:
What should we do if the `Update::updateToCurrentVersion();` is already in progress?
With this code we have an uncaught `Concrete\Core\System\Mutex\MutexBusyException` exception.

Reply:
Can we make it so that the maintenance mode view is rendered if this is the case?

Reply:
Sure! Here it is: https://github.com/concrete5/concrete5/pull/6480